README.md

```diff
@@ -1,28 +1,123 @@
+## Introduction
 
-## Usage
+This is a web application that will extract large amounts of data from the Google search results page.
 
-Clone the repository
+## Project Setup
 
-`git clone git@github.com:nimblehq/git-template.git`
+### Prerequisites
 
-## License
+- Ruby version: 2.7.2
+- Node version: 14.15.4
 
-This project is Copyright (c) 2014 and onwards. It is free software,
-and may be redistributed under the terms specified in the [LICENSE] file.
+### Docker
 
-[LICENSE]: /LICENSE
+- Install [Docker for Mac](https://docs.docker.com/docker-for-mac/install/)
 
-## About
+- Setup and boot the Docker containers:
 
-![Nimble](https://assets.nimblehq.co/logo/dark/logo-dark-text-160.png)
+```sh
+./bin/envsetup.sh
+```
 
-This project is maintained and funded by Nimble.
+### Development
 
-We love open source and do our part in sharing our work with the community!
-See [our other projects][community] or [hire our team][hire] to help build your product.
+- Setup the databases:
 
-[community]: https://github.com/nimblehq
-[hire]: https://nimblehq.co/
+  - Postgres:
+
+  ```sh
+  rake db:setup
+  ```
+
+- Run the Rails app
+
+```sh
+foreman start -f Procfile.dev
+```
+
+## Tests
+
+### Docker-based tests on the CI server
+
+Add the following build settings to run the tests in the Docker environment via Docker Compose (configuration in `docker-compose.test.yml`):
+
+- Configure the environment variable `BRANCH_TAG` to tag Docker images per branch, e.g:
+
+```sh
+# a unique `BRANCH_TAG` value to tag the Docker image
+# e.g $SEMAPHORE_BRANCH_ID or using the
+# or using nimblehq/branch-tag-action@v1.2 Github action
+export BRANCH_TAG= # unique value for tagging Docker image
+```
+
+Each branch needs to have its own Docker image to avoid build settings disparities and leverage Docker image caching.
+
+> BRANCH_TAG must not contain special characters (`/`) to be valid. So using $BRANCH_NAME will not work e.g. chore/setup-docker.
+An alternative is to use a unique identifier such as PR_ID or BRANCH_ID on the CI server.
+
+- Pull the latest version the Docker image for the branch:
+
+```sh
+docker pull $DOCKER_IMAGE:$BRANCH_TAG || true
+```
+
+On each build, the CI environment does not contain yet a cached version of the image. Therefore, it is required to pull
+it first to leverage the `cache_from` settings of Docker Compose which avoids rebuilding the whole Docker image on subsequent test builds.
+
+- Build the Docker image:
+
+```sh
+./bin/docker-prepare && docker-compose -f docker-compose.test.yml build
+```
+
+Upon the first build, the whole Docker image is built from the ground up and tagged using `$BRANCH_TAG`.
+
+- Push the latest version of the Docker image for this branch:
+
+```sh
+docker push $DOCKER_IMAGE:$BRANCH_TAG
+```
+
+- Setup the test database:
+
+```sh
+docker-compose -f docker-compose.test.yml run test bin/bundle exec rake db:test:prepare
+```
+
+#### Semaphore CI 2.0
```

Comment:
As we will not use Semaphore for CI/CD, I think we can remove it

Reply:
Good catch!

Reply:
@junan Updated in [`afccc22` (#26)](https://github.com/suho/google-search-ruby/pull/26/commits/afccc223fe6d43e653da5aa94bc7f0abf31298c5)